dexios/src/subcommands/header_key.rs

```diff
@@ -91,18 +92,14 @@ pub fn change_key(input: &str, key_old: &Key, key_new: &Key) -> Result<()> {
 
             let master_key_result =
                 cipher.decrypt(&keyslot.nonce, keyslot.encrypted_key.as_slice());
-            let mut master_key_decrypted = master_key_result.map_err(|_| {
+            let master_key_decrypted = master_key_result.map_err(|_| {
                 anyhow::anyhow!(
                     "Unable to decrypt your master key (maybe you supplied the wrong key?)"
                 )
             })?;
 
-            let mut master_key = [0u8; MASTER_KEY_LEN];
-            let len = MASTER_KEY_LEN.min(master_key_decrypted.len());
-            master_key[..len].copy_from_slice(&master_key_decrypted[..len]);
-
-            master_key_decrypted.zeroize();
-            let master_key = Protected::new(master_key);
+            let master_key =
+                Protected::<[u8; MASTER_KEY_LEN]>::new(vec_to_arr(master_key_decrypted));
```

Comment:
Here we can write less annotations

```rust
Protected::new(vec_to_arr::<MASTER_KEY_LEN>(master_key_descrypted));
```

Reply:
Ah, I must have missed this one! I'll sort it when I can :)